step-functions/audit_event.asl.json

```diff
@@ -1,248 +1,257 @@
-				"component_id.$": "$$.Execution.Input.detail.issuer",
-				"user.$": "$$.Execution.Input.detail.user"
-			},
-			"ResultPath": "$[0].auditEvent",
-			"OutputPath": "$",
-			"Next": "Is evidence Present?"
-		},
-		"Add Restricted Info to AuditEvent": {
-			"Type": "Pass",
-			"Comment": "The user context info is returned in added with the Restricted user information",
-			"Parameters": {
-				"event_name.$": "States.Format('{}_{}', $[0].prefix, $[0].type)",
-				"timestamp.$": "$[1].epochSeconds.value",
-				"event_timestamp_ms.$": "$[2].epochMilliseconds.value",
-				"component_id.$": "$$.Execution.Input.detail.issuer",
-				"user.$": "$$.Execution.Input.detail.user",
-				"restricted.$": "States.JsonMerge($[0].restricted.value, $[0].formattedDeviceInformation, false)"
-			},
-			"ResultPath": "$[0].auditEvent",
-			"OutputPath": "$",
-			"Next": "Is evidence Present?"
-		},
-		"Is evidence Present?": {
-			"Type": "Choice",
-			"Comment": "This check if the input Details has an evidence to add to the audit structure",
-			"Choices": [
-				{
-					"Variable": "$$.Execution.Input.detail.evidence",
-					"IsPresent": true,
-					"Next": "Audit Event With Evidence & Restricted Info"
-				}
-			],
-			"Default": "Add User Context"
-		},
-		"Add User Context": {
-			"Type": "Pass",
-			"Parameters": {
-				"auditEvent.$": "$.[0].auditEvent"
-			},
-			"ResultPath": "$[0]",
-			"OutputPath": "$",
-			"Next": "publish event to TxMa Queue"
-		},
-		"Audit Event With Evidence & Restricted Info": {
-			"Type": "Pass",
-			"Comment": "Merges the evidence detail contained in the details input by interpolating it into a stringified version of {evidence: {}}, {} is replaced with string form of the incoming evidence detail, which is then converted to Json and merge with the auditEvent state",
-			"Parameters": {
-				"auditEvent.$": "States.JsonMerge($.[0].auditEvent, States.StringToJson(States.Format('\\{\"extensions\":\\{\"evidence\": {}\\}\\}', States.JsonToString($$.Execution.Input.detail.evidence))), false)"
-			},
-			"ResultPath": "$[0]",
-			"OutputPath": "$",
-			"Next": "publish event to TxMa Queue"
-		},
-		"publish event to TxMa Queue": {
-			"Type": "Task",
+  "Comment": "The Audit Event State Machine, responsible for generating the overall event structure",
+  "StartAt": "Get Audit name components",
+  "States": {
+    "Get Audit name components": {
+      "Type": "Pass",
+      "Comment": "The details JSON object is stringified using States.JsonToString and then split it on ':\"' creating an array with key and value as items if a key called $$.Execution.Input.detail is found deviceInformation returns boolean value true",
+      "Parameters": {
+        "prefix.$": "$$.Execution.Input.detail.auditPrefix",
+        "type.$": "$$.Execution.Input.detail-type",
+        "deviceInformation.$": "States.ArrayContains(States.StringSplit(States.JsonToString($$.Execution.Input.detail), ':\"'), 'deviceInformation')"
```

Comment:
So this is actually checking the `Execution.Input.detail`, right? And then we check the `deviceInformation` in the `Is Device Information Value Present?` step? What do you think about renaming this to something like `inputDetail`, then we can reuse it for other things in the future if needed

Reply:
Could you elaborate please? @Caitlin-cooling

Reply:
Sorry what is it that you want me to elaborate on? Just wondering if `deviceInformation` could be renamed to something like `inputDetail` to better represent what it actually is and make it more reusable in future?

Reply:
1. So this `deviceInformation` is a boolean, true if `deviceInformation` is present. 
2. It is then used by `Is Device Information Value Present?` to control the flow of the state machine.

`$$.Execution.Input.detail` is global so it will be available to all states. 

I'm not sure renaming it `inputDetail` will help? Apologies if I am not understanding, brain is slow today :(

Reply:
Ignore me I understand now 🤦‍♀️